oauth2_http/java/com/google/auth/oauth2/ServiceAccountJwtAccessCredentials.java

```diff
@@ -115,21 +123,29 @@ private ServiceAccountJwtAccessCredentials(
    * @param privateKey RSA private key object for the service account.
    * @param privateKeyId Private key identifier for the service account. May be null.
    * @param defaultAudience Audience to use if not provided by transport. May be null.
+   * @param universeDomain universe domain in the format some-domain.xyz. By default, returns
+   *     googleapis.com.
    */
   private ServiceAccountJwtAccessCredentials(
       String clientId,
       String clientEmail,
       PrivateKey privateKey,
       String privateKeyId,
       URI defaultAudience,
-      String quotaProjectId) {
+      String quotaProjectId,
+      String universeDomain) {
     this.clientId = clientId;
     this.clientEmail = Preconditions.checkNotNull(clientEmail);
     this.privateKey = Preconditions.checkNotNull(privateKey);
     this.privateKeyId = privateKeyId;
     this.defaultAudience = defaultAudience;
     this.credentialsCache = createCache();
     this.quotaProjectId = quotaProjectId;
+    if (universeDomain == null || universeDomain.trim().isEmpty()) {
```

Comment:
Can this validation be done in `setUniverseDomain()`? If user passes null or empty universe_domain, then we throw an IllegalArgumentException

Reply:
From offline discussion: We've decided to keep this logic in the constructor. This follows the convention set by ServiceAccountCredentials which sets the default value to GDU if empty. See https://github.com/googleapis/google-auth-library-java/blob/b347603db4deabb4eb34ed14b96178c95d3e7b45/oauth2_http/java/com/google/auth/oauth2/ServiceAccountCredentials.java#L127

where super is GoogleCredentials and sets the universe_domain in https://github.com/googleapis/google-auth-library-java/blob/b347603db4deabb4eb34ed14b96178c95d3e7b45/oauth2_http/java/com/google/auth/oauth2/GoogleCredentials.java#L355-L358

This logic is slightly hidden since ServiceAccountCredentials takes in a Builder as a parameter to the constructor whereas ServiceAccountJwtCredentials takes in individual fields